imblearn/over_sampling/tests/test_rose.py

```diff
@@ -0,0 +1,59 @@
+"""Test the module ROSE."""
+# Authors: Andrea Lorenzon <andrelorenzon@gmail.com>
+# License: MIT
+
+import numpy as np
+
+from imblearn.over_sampling import ROSE
+
+
+def test_testunit():
+    """Check test unit"""
+    return True
+
+
+def test_random_state():
+    """Check randomState()"""
+    assert np.random.RandomState(42)
+
+
+def test_instance():
+    """Check ROSE instantiation"""
+    rose = ROSE()
+    assert id(rose)
+
+
+RND_SEED = 0
+X = np.array(
+    [
+        [0.11622591, -0.0317206],
+        [0.77481731, 0.60935141],
+        [1.25192108, -0.22367336],
+        [0.53366841, -0.30312976],
+        [1.52091956, -0.49283504],
+        [-0.28162401, -2.10400981],
+        [0.83680821, 1.72827342],
+        [0.3084254, 0.33299982],
+        [0.70472253, -0.73309052],
+        [0.28893132, -0.38761769],
+        [1.15514042, 0.0129463],
+        [0.88407872, 0.35454207],
+        [1.31301027, -0.92648734],
+        [-1.11515198, -0.93689695],
+        [-0.18410027, -0.45194484],
+        [0.9281014, 0.53085498],
+        [-0.14374509, 0.27370049],
+        [-0.41635887, -0.38299653],
+        [0.08711622, 0.93259929],
+        [1.70580611, -0.11219234],
+    ]
+)
+Y = np.array([0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0])
+R_TOL = 1e-4
+
+
+def test_rose():
+    """Check ROSE use"""
+    X_res, y_res = ROSE().fit_resample(X, Y)
+    assert np.unique(Y.all()) == np.unique(y_res.all())
+    assert X_res.shape[1] == X.shape[1]
```

Comment:
`X`, `Y`, `RTOL` should be reworked as local variables in `test_rose`

e.g. see this test for SMOTE:

https://github.com/scikit-learn-contrib/imbalanced-learn/blob/0acd717578672e5d910107e81b5fb80830de54ea/imblearn/over_sampling/tests/test_smote.py#L47-L107

Reply:
Variables has been inserted in the pertinent function definition.
Unused variables has been removed.

NOTE: I foresee that I will expand test units, and try to add multiclass oversampling capabilities, after this iteration, so i'll have to change both ROSE and the test units.